packages/protobuf-encoder/index.js

```diff
@@ -0,0 +1,45 @@
+import msgpack from '@ygoe/msgpack'
+
+import { action_cable as protos } from './generated/message_pb.js'
+
+const Message = protos.Message
+const MessageType = protos.Type
+const Command = protos.Command
+
+export class ProtobufEncoder {
+  encode(msg) {
+    // convert enum value names to corresponding integers
+    msg.command = Command[msg.command]
+    msg.type = MessageType[msg.type]
+
+    if (msg.message !== undefined) {
+      msg.message = msgpack.serialize(msg.message)
+    }
+
+    let err = Message.verify(msg)
+
+    if (err) {
+      throw Error(err)
+    }
+
+    return Message.encode(msg).finish()
+  }
+
+  /* eslint-disable consistent-return */
+  decode(data) {
+    try {
+      let decodedMessage = Message.decode(data)
+
+      if (decodedMessage.type !== undefined) {
+        let messageTypesById = Object.getPrototypeOf(MessageType)
```

Comment:
That's pretty ugly 😕 Enum object prototypes hold inverse hashes of enum values (e.g. if enum is `{ foo: 1}`, then its prototype will be `{1: foo}`). I'll try to find a better solution to this

Reply:
Well, I didn't find anything better than just wrap this logic into separate class /cc @palkan